[EISW-141730] 19.x Merge dialect changes

sramasit opened the pull request.

ArtemySkrebkov commented on mlir/include/mlir/Dialect/CMakeLists.txt:
```diff
@@ -44,4 +44,3 @@ add_subdirectory(UB)
 add_subdirectory(Utils)
 add_subdirectory(Vector)
 add_subdirectory(X86Vector)
```
This looks a bit radical. Is it possible to introduce a cmake option for disabling the dialect?

Something like we have for spirv:
https://github.com/intel/npu-plugin-llvm/blob/bb62f0284f6fe478222256a4e02e4b868f6b0eb2/mlir/include/mlir/Dialect/CMakeLists.txt#L37

sramasit replied:
I see. Based on the comment in the JIRA I thought we are doing a permanent cleanup. My initial solution was to disable using CMake options like you pointed above. I will proceed with CMake option then.

sramasit replied:
Done

sramasit committed:
EISW-163563 Add CMake option to conditionally enable XeGPU dialect

Introduced a CMake option MLIR_DIALECT_XEGPU_ENABLE (default OFF) to optionally include the XeGPU dialect in the build.

Signed-off-by: Surendar Rama Sitaraman <surendar.rama.sitaraman@intel.com>

andrey-golubev reviewed (changes requested):
Can we bring this into the upstream first? And start a general discussion on the granularity of the MLIR? Otherwise, this is going to live forever in our repo which has a maintenance cost.

sramasit committed:
EISW-163564 Add CMake option to conditionally disable TosaToTensor conversion

nikita-kud commented on mlir/include/mlir/Support/Timing.h:
```diff
@@ -17,7 +17,6 @@
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/StringMapEntry.h"
 #include "llvm/Support/raw_ostream.h"
-#include "llvm/Support/Format.h"
```
Could please open a PR only for this commit?

sramasit replied:
Done
